output/cloud/expv2/mapping.go

```diff
@@ -10,21 +11,41 @@ import (
 )
 
 // TODO: unit test
-func mapTimeSeriesLabelsProto(tags *metrics.TagSet) []*pbcloud.Label {
+func mapTimeSeriesLabelsProto(tags *metrics.TagSet) ([]*pbcloud.Label, []string) {
 	labels := make([]*pbcloud.Label, 0, ((*atlas.Node)(tags)).Len())
+	var discardedLabels []string
 
-	// TODO: move it as a shared func
+	// TODO: move this as a shared func
 	// https://github.com/grafana/k6/issues/2764
 	n := (*atlas.Node)(tags)
 	if n.Len() < 1 {
-		return labels
+		return labels, discardedLabels
 	}
 	for !n.IsRoot() {
 		prev, key, value := n.Data()
-		labels = append(labels, &pbcloud.Label{Name: key, Value: value})
+
 		n = prev
+		if !isReservedLabelName(key) {
+			labels = append(labels, &pbcloud.Label{Name: key, Value: value})
+			continue
+		}
+
+		if discardedLabels == nil {
+			discardedLabels = make([]string, 0, 1)
+		}
+
+		discardedLabels = append(discardedLabels, key)
 	}
-	return labels
+	return labels, discardedLabels
+}
+
+func isReservedLabelName(name string) bool {
+	// this is a reserved label prefix for the prometheus
+	if strings.HasPrefix(name, "__") {
```

Comment:
```suggestion
	if strings.HasPrefix(name, "__") && strings.HasSuffix("__") {
```
In theory, `__my_amazing_label_` should be a legit tag key. Should we care about it?

Reply:
[According convention](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels:~:text=Label%20names%20beginning%20with%20__%20(two%20%22_%22)%20are%20reserved%20for%20internal%20use) it's not since it begins with `__`

Reply:
Yeah, you're right. I should double-check one more the next time. :pray: